src/main/java/org/elasticsearch/indices/query/IndicesQueriesModule.java

```diff
@@ -30,30 +30,21 @@
 
 public class IndicesQueriesModule extends AbstractModule {
 
-    private Set<Class<QueryParser>> queryParsersClasses = Sets.newHashSet();
-    private Set<QueryParser> queryParsers = Sets.newHashSet();
+    private Set<Class<? extends QueryParser>> queryParsersClasses = Sets.newHashSet();
 
-    public synchronized IndicesQueriesModule addQuery(Class<QueryParser> queryParser) {
+    public synchronized IndicesQueriesModule addQuery(Class<? extends QueryParser> queryParser) {
         queryParsersClasses.add(queryParser);
         return this;
     }
 
-    public synchronized IndicesQueriesModule addQuery(QueryParser queryParser) {
```

Comment:
I typically defaulted to this method because it avoids the generics gymnastics I mentioned above and I'd use the Class version when I needed guice to build the query. Since you've fixed the signature above I don't see a good reason for this. If you need the global state then you can register the extra classes with guice elsewhere and have them injected.